.github/workflows/docs-ci-prod.yaml

```diff
@@ -0,0 +1,38 @@
+# .github/workflows/preview.yml
+name: "docs.estuary.dev: Deploy"
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - site/**
+
+concurrency: docs-${{ github.ref }}
+
+permissions:
+  contents: write
+
+jobs:
+  deploy-docs:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+          cache-dependency-path: ./site/package-lock.json
+      - name: Install dependencies
+        working-directory: ./site
+        run: npm ci
+      - name: Build `docs.estuary.dev`
+        working-directory: ./site
+        run: npm run build
+      - name: Deploy preview
+        uses: JamesIves/github-pages-deploy-action@v4
```

Comment:
Why this instead of https://github.com/marketplace/actions/deploy-github-pages-site?

Reply:
I saw that, but two things seemed weird:
* The [readme says](https://github.com/marketplace/actions/deploy-github-pages-site#scope) 
  > ⚠️ Official support for building Pages with Actions is in public beta at the moment.
* And I wasn't clear whether it would work with the PR preview deploys. The [action](https://github.com/marketplace/actions/deploy-pr-preview) I used for that suggests using this one from `JamesIves`, so that's the one I went with.

Thoughts?